### DISCUSSION TITLE
question about setup function

### DISCUSSION THREAD

**Discussion by kangaroolab** [treehousedev/treehouse#312] [General]
for the setup function:

`export async function setup(document: Document, target: HTMLElement, backend: Backend)`

the first argument is Document type but seems we are passing window.document. is it the same?

is there anyway we can customize setup e.g. add new command/menu or even open new panel as default?

**Comment by progrium** [comment]
I guess we could break down setup so you can more easily build your own. Yes, `window.document` is a `Document`. Let me know if that resolves this issue!
